src/dir_htree.rs

```diff
@@ -377,12 +377,9 @@ pub(crate) fn get_dir_entry_via_htree(
             inode.index,
             path.clone(),
         )?;
-        offset_within_block =
-            if let Some(offset) = offset_within_block.checked_add(entry_size) {
-                offset
-            } else {
-                return Err(Corrupt::DirEntry(inode.index.get()).into());
-            };
+        offset_within_block = offset_within_block
+            .checked_add(entry_size)
+            .ok_or(Ext4Error::Corrupt(Corrupt::DirEntry(inode.index.get())))?;
```

Comment:
```
        offset_within_block = offset_within_block
            .checked_add(entry_size)
            .ok_or(Corrupt::DirEntry(inode.index.get()))?;
```
seems to work on my end?

Reply:
You are right, thanks for checking that. I incorrectly still had an `into()` in there, which isn't needed and made the compiler confused.